pkg/parser/uri.go

```diff
@@ -0,0 +1,32 @@
+package parser
+
+import (
+	"errors"
+	"net/url"
+)
+
+type parsedURI struct {
```

Comment:
Do we really need this extra struct? Why not use simply `url.URL` instead?

Reply:
Ah, simpler. Done.